### PR TITLE
add -a and -D (asinfo lookups)

### PR DESCRIPTION
**Pull request by vixie**
joe, daniel, and henry have verified functionality.

### CODE REVIEW

File: asinfo.c [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+static struct __res_state res;
+
+static const char *asinfo_from_ipv4(const char *, char **, char **);
+#ifdef asinfo_ipv6
+static const char *asinfo_from_ipv6(const char *, char **, char **);
+#endif
+static const char *asinfo_from_dns(const char *, char **, char **);
+static const char *keep_best(char **, char **, char *, char *);
+
+const char *
```
Comment (djw1149) [inline asinfo.c:40]: add function comment

File: asinfo.c [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+#include <resolv.h>
+#include <stdlib.h>
+#include <string.h>
+#include "asinfo.h"
+#include "globals.h"
+
+static struct __res_state res;
+
+static const char *asinfo_from_ipv4(const char *, char **, char **);
+#ifdef asinfo_ipv6
```
Comment (djw1149) [inline asinfo.c:34]: explain why we need asinfo_ipv6.

File: asinfo.c [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+		strcpy(p, asinfo_domain);
+		result = asinfo_from_dns(dname, asnum, cidr);
+	}
+	p = NULL;
+	free(dname);
+	return result;
+}
+#endif
+
+static const char *
```
Comment (djw1149) [inline asinfo.c:118]: explain what we expect to be parsing.
why TXT>3 is an error because it doesn't match that expectation.

**Review by djw1149 — CHANGES_REQUESTED** [review]
I'd like function headers added in asinfo.c

File: asinfo.c (hunk 3 above)
Comment (vixie) [reply asinfo.c:160]: To github.com:dnsdb/dnsdbq.git
   15c743d..36e9a8f  asnlookup -> asnlookup

File: asinfo.c (hunk 1 above)
Comment (vixie) [reply asinfo.c:54]: To github.com:dnsdb/dnsdbq.git
   1fbef16..15c743d  asnlookup -> asnlookup

File: asinfo.c (hunk 2 above)
Comment (vixie) [reply asinfo.c:40]: see also:

To github.com:dnsdb/dnsdbq.git
   1fbef16..15c743d  asnlookup -> asnlookup

**Review by djw1149 — APPROVED** [review]

**Pull request merged**